### ISSUE TITLE
TensorFlow 1.8.0 build is ~4x slower on V100 GPU than upstreams wheel

### ISSUE THREAD

**Issue by akesandgren** [easybuilders/easybuild-easyconfigs#6515]
The upstream wheel of TF 1.8.0 for GPU is ~4x faster than the one EB builds from source on a Skylake Gold with a Volta V100.
Tested with models/tutorials/image/cifar10/ from git@github.com:tensorflow/models.git

This should be looked at more closely.

**Comment by boegel** [comment]
@akesandgren Which CUDA compute capabilities were used?

@wpoely86 Can you test this on P100?

**Comment by wpoely86** [comment]
How and what exactly did you run from `cifar10`?

**Comment by wpoely86** [comment]
Single P100:
- With docker image with tag 1.8.0-py3-gpu: 989 s
- With EB version: 963 s

EB version is TF 1.8 with CUDA 9.2.88 (driver 396.26)  + cuDNN/7.1.4.18

Single run of `cifar10_train.py`

**Comment by boegel** [comment]
@wpoely86 Which exact easyconfig file is that for the EB installation?

**Comment by wpoely86** [comment]
The same one as in #6412 but not using fosscuda.

**Comment by akesandgren** [comment]
On a 28 core Skylake Gold 6132 CPU with dual Volta V100 cards I was running on 14 cores and a single V100 card:
```
export OMP_NUM_THREADS=14
parms="--inter_op_parallelism_threads=1 --intra_op_parallelism_threads=$OMP_NUM_THREADS --batch_size=128"
steps=10000
for fp16 in False ; do
    echo Running $steps steps with fp16=$fp16
    time python cifar10_train.py $parms --max_steps=$steps --use_fp16=$fp16
    echo ""
done
```

**Comment by Sneeg** [comment]
If it's relevant, I have both a Tesla V100 and a Quadro GV100 at my hands and could not reproduce this on any of them. The host was a dual socket Haswell system, though.
However, I have noticed that a AVX2+FMA build is almost always a bit slower (few percent) than a SSE-only version most of the time.

**Comment by wpoely86** [comment]
I'm will try to post the specifics on Monday but I'm not seeing any serious performance difference between EB and the docker images.

Which nvidia driver are you using? Did you try with Cuda 9.2?

**Comment by akesandgren** [comment]
The binary whls requires cuda 9.0 so I'm comparing with an EB build using Cuda 9.0
And i'm not using the docker images, i'm using the binary whl from tensorflow directly, i.e. doing the install of TF that was in EB before we got the build-from-source.

I.e. I have a Tensorflow module (note lower case f) which is the binary whl install.
And i have only checked on our V100 cards, not on the K80s

**Comment by akesandgren** [comment]
This is what i get:
(The first one is the EB build-from-source, the second one the binary whell install)

```
Using fosscuda/2018a and TensorFlow/.1.9.0-rc1-Python-3.6.4
python cifar10_train.py --inter_op_parallelism_threads=1 --intra_op_parallelism_threads=14 --batch_size=128 --max_steps=10000 --use_fp16=False
2018-07-07 07:30:45.516446: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties: 
name: Tesla V100-PCIE-16GB major: 7 minor: 0 memoryClockRate(GHz): 1.38
...
2018-07-07 07:34:59.208041: step 9990, loss = 0.86 (5566.7 examples/sec; 0.023 s
ec/batch)

real    4m25.964s
user    13m17.616s
sys     2m31.608s


Using goolfc/2017b and Tensorflow/.1.8.0-Python-3.6.3
2018-07-07 07:35:11.578585: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2018-07-07 07:35:11.930816: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties: 
name: Tesla V100-PCIE-16GB major: 7 minor: 0 memoryClockRate(GHz): 1.38
...
2018-07-07 07:36:34.532438: step 9990, loss = 0.76 (23441.4 examples/sec; 0.005 sec/batch)

real    1m34.320s
user    7m34.440s
sys     1m35.280s
```

**Comment by akesandgren** [comment]
The nvidia driver in use is 396.24

**Comment by boegel** [comment]
@akesandgren Any updates on this?

**Comment by akesandgren** [comment]
No, I haven't had time to look at it after the vacation. I hope to get somewhere later on, but comparing v1.10

**Comment by boegel** [comment]
@akesandgren ping?

**Comment by boegel** [comment]
@akesandgren I believe you have some updates on this?

**Comment by surak** [comment]
JSC has seen the same. TF 1.12 is around 4x faster here

**Comment by boegel** [comment]
@surak Which toolchain was being used here? Also on V100?

**Comment by surak** [comment]
Stages/Devel-2018a  GCC/5.5.0  MVAPICH2/2.3a-GDR

There was a version compiled with Intel and ParaStationMPI, but I never bothered.

Yes, with V100. The speed difference on the K80s was negligible.

**Comment by boegel** [comment]
There are a couple of important environment variables that should be set to get good performance (we noticed this too recently).
The pre-built wheels may have different defaults baked in, which could explain the performance differences...
See the `Tuning MKL for the best performance` section in https://www.tensorflow.org/guide/performance/overview (which is relevant regardless of whether or not you're using Intel MKL, since it also applies to `mkl-dnn`).
